app/models/subject.rb

```diff
@@ -22,5 +22,35 @@ def approved?(approved_approvable_ids)
     approved_approvable_ids.include?(exam ? exam.id : course.id)
   end
 
+  def category
+    if semester.present?
+      case semester
+      when 1 then :first_semester
+      when 2 then :second_semester
+      when 3 then :third_semester
+      when 4 then :fourth_semester
+      when 5 then :fifth_semester
+      when 6 then :sixth_semester
+      when 7 then :seventh_semester
+      when 8 then :eighth_semester
+      when 9 then :nineth_semester
+      end
+    elsif !active?
+      :inactive
+    elsif revalid?
+      :revalid
+    elsif external?
+      :external
+    elsif extension_module?
+      :extension_module
+    else
+      :optional
+    end
+  end
```

Comment:
We could also avoid creating one attribute per each category and just add an enum attribute `category`.

Thoughts about using that approach instead?